codegen/src/main/twirl/templates/JavaClient/Client.scala.txt

```diff
@@ -36,14 +36,14 @@ public abstract class @{service.name}Client extends @{service.name}ClientPowerAp
       private final Materializer mat;
       private final ExecutionContext ec;
 
-      private Default@{service.name}Client(GrpcClientSettings settings, ClassicActorSystemProvider sys) {
+      private Default@{service.name}Client(GrpcClientSettings settings, ClassicActorSystemProvider sys,  ExecutionContext ec) {
```

Comment:
I'm not sure customizing the executioncontext is sufficient here: a lot of the gRPC machinery is streams-based, and will use the system materializer. I didn't check, but I think the system materializer also runs on the default dispatcher.

In any case I think it should be an optional parameter, so we should also provide a version of the constructors that only takes the `sys`.

Reply:
@raboof Yeah I was making a grpc call from a persistent actor and got some thread starvation when we were load testing it. However instead of generating the grpc client I wrote them by hand as the generator and this time pass in a different executor context (I mean the scala version) instead of the main actor system dispatcher and things were pretty smooth.

Reply:
Also I think if we can specify a different materializer for the client maybe that can help a bit.

Reply:
> I think if we can specify a different materializer for the client maybe that can help a bit. That materializer will be bound to the lifecycle the akka grpc client.

Hmm, so instead of using the system materializer or letting the client provide their own materializer, you're suggesting creating a dedicated materializer 'inside' the client, tied to its lifecycle? That might be interesting.